src/hash.jl

```diff
@@ -23,12 +23,54 @@ mutable struct TextHashFunction
     cardinality::Int
 end
 
+"""
+```
+TextHashFunction(cardinality)
+TextHashFunction(hash_function, cardinality)
+```
+
+The need to create a lexicon before we can construct a document term matrix is often prohibitive.
+We can often employ a trick that has come to be called the Hash Trick in which we replace terms
+with their hashed valued using a hash function that outputs integers from 1 to N.
+
+Parameters:
+	-  cardinality	    = Max index used for hashing (default 100)
+ 	-  hash_function    = function used for hashing process (default function present, see code-base)
+
+```julia-repl
+julia> h = TextHashFunction(10)
+TextHashFunction(hash, 10)
+```
+"""
 TextHashFunction(cardinality::Int) = TextHashFunction(hash, cardinality)
 
 TextHashFunction() = TextHashFunction(hash, 100)
 
 cardinality(h::TextHashFunction) = h.cardinality
 
+"""
+```
+index_hash(str, TextHashFunc)
+```
+
+Shows mapping of string to integer.
+
+Parameters:
+	-  str		   = Max index used for hashing (default 100)
+ 	-  TextHashFunc    = TextHashFunction type object
```

Comment:
Is this necessary? Doesn't `str` imply String and `TextHashFunc` imply TextHashFunction?